querysrv.go

```diff
@@ -329,10 +329,12 @@ func (s *querysrv) handleAnnounce(ctx context.Context, remote net.IP, deviceID p
 		}
 
 		ip := net.ParseIP(host)
-		if len(ip) == 0 || ip.IsUnspecified() {
-			uri.Host = net.JoinHostPort(remote.String(), port)
+		if host == "" || ip.IsUnspecified() {
```

Comment:
But `net.ParseIP(host)` returns `nil` for FQDN's as well, becuase they are not IP's. The `if` condition is specific for empty host names `xxx://:port` and unspecified IP addresses `xxx://0.0.0.0:port`. Any other value for the host part will be kept as-is.